lib/thumbs/pull_request_worker.rb

```diff
@@ -69,19 +68,34 @@ def try_merge
 
       status={}
       status[:started_at]=DateTime.now
-      cleanup_build_dir
+
+
+     debug_message "Trying merge #{pr.base.repo.full_name}:PR##{pr.number} \" #{pr.title}\" #{pr.head.repo.full_name}##{most_recent_head_sha} onto #{@pr.base.ref} #{most_recent_base_sha}"
       begin
-        git = clone(@build_dir)
-        git.checkout(most_recent_head_sha)
-        git.checkout(@pr.base.ref)
+        git = refresh_repo
+        git.reset
+        git.checkout(pr.base.ref)
         git.branch(pr_branch).checkout
         debug_message "Trying merge #{@repo}:PR##{@pr.number} \" #{@pr.title}\" #{most_recent_head_sha} onto #{@pr.base.ref} #{most_recent_base_sha}"
-        merge_result = git.merge("#{most_recent_head_sha}")
+
+        if pr.base.repo != pr.head.repo
+          contributor_repo="git://github.com/#{pr.head.repo.full_name}"
+          debug_message("forked branch pr Contributor REPO: #{contributor_repo}")
+          remotes=git.remotes.collect{|r| r.name }
+          unless remotes.include?("contributor")
```

Comment:
If I understand this correctly, we are able to use "contributor" instead of the actual contributor's handle here b/c the local git repository is temporary. If this invariant does not hold, we should use the contributor's handle instead.